presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -235,7 +236,17 @@ protected RelationPlan visitJoin(Join node, Void context)
                     continue;
                 }
 
-                if (conjunct instanceof ComparisonExpression) {
+                Set<QualifiedName> dependencies = DependencyExtractor.extractNames(conjunct, analysis.getColumnReferences());
+                boolean isJoinUsing = node.getCriteria().filter(JoinUsing.class::isInstance).isPresent();
+                if (!isJoinUsing && (dependencies.stream().allMatch(left::canResolve) || dependencies.stream().allMatch(right::canResolve))) {
+                    // If the conjunct can be evaluated entirely with the inputs on either side of the join, add
+                    // it to the list complex expressions and let the optimizers figure out how to push it down later.
+                    // Due to legacy reasons, the expression for "join using" looks like "x = x", which (incorrectly)
+                    // appears to fit the condition we're after. So we skip them.
```

Comment:
> / Due to legacy reasons, the expression for "join using" looks like "x = x", which (incorrectly) appears to fit the condition we're after.

Can you please post an issue about this. I think it would be nice to fix that also as it unnecessarily complicates codes of planner. IMHO `x JOIN y USING c` should be rewritten to `x JOIN y ON x.c = y.c`. That way planner would only care of one form of join criteria, that would simplify the code.